app/models/carrier.rb

```diff
@@ -24,4 +31,14 @@ def api_key
   def webhooks_enabled?
     webhook_endpoint&.enabled?
   end
+
+  def good_standing?
+    return false unless enabled? || restricted?
+
+    !restricted? || remaining_interactions.positive?
+  end
```

Comment:
Because there are few methods that are related in checking the carreir in good standing, so better to move it to its own object.